superset-frontend/src/SqlLab/components/TemplateParamsEditor.jsx

```diff
@@ -95,53 +91,56 @@ export default class TemplateParamsEditor extends React.Component {
         syntax.
       </p>
     );
-  }
+    return renderDocContent;
+  };
 
-  renderModalBody() {
-    return (
+  const renderModalBody = () => {
+    const renderModalBodyContent = (
       <div>
-        {this.renderDoc()}
+        {renderDoc()}
         <StyledConfigEditor
           keywords={[]}
-          mode={this.props.language}
+          mode={props.language}
           minLines={25}
           maxLines={50}
-          onChange={this.onChange}
+          onChange={onChange}
           width="100%"
           editorProps={{ $blockScrolling: true }}
           enableLiveAutocompletion
-          value={this.state.codeText}
+          value={setState(state.codeText)}
```

Comment:
why's the value the result of the setState call here? That doesn't seem quite right